ENIGMAsystem/SHELL/Platforms/General/POSIX/Utility.cpp

```diff
@@ -103,6 +105,24 @@ unsigned long get_timer() {  // microseconds since the start of the game
   return (enigma::time_current.tv_sec) * 1000000 + (enigma::time_current.tv_nsec / 1000);
 }
 
+std::string shellscript_evaluate(std::string command) {
+  char *buffer = NULL;
+  size_t buffer_size = 0;
+  std::string result = "";
+
+  FILE *file = popen(command.c_str(), "r");
+  while (getline(&buffer, &buffer_size, file) != -1)
+    result += buffer;
+
+  free(buffer);
+  pclose(file);
+
+  if (result.back() == '\n')
```

Comment:
I don't think modifying the output is wise. If there was no newline in the output, `fgets`/`getline` will not add one.

Reply:
well from my testing it *does* add a new line. I suppose this bullshit is an OS bug too? Also happens on Mac.... .. ... .

Reply:
i know the Linux man page claims it doesn't add one if it isn't found but i think it is not referring to the exclusion of the end of file where yes there is always a new line. As Robert said, that's how DOS and Unix generally works.

Reply:
A lot of people do not typically like a random new line at the end of a string return value when it isn't asked for.

Reply:
I will be re-using this as a helper function with the new linux widget systems re-writes. So it is important that is does everything i need it to within reason

Reply:
@JoshDreamland could you tell me what purpose it serves to have a new line at the end, whether the function is responsible for it or otherwise?

Reply:
No OS bugs, here. The newline is annoying in some programs, and crucial in others. I believe we already have a `string_strip` utility method, and if not, we could really use one. Users who didn't want to deal with extra whitespace would just pass the result to `string_strip()`, while users who needed it (eg, as a delimiter) would use it however they needed.

Perhaps the circumstances where the newline is important are rare, particularly in game design (where the need for this function is, itself, rare), but you writing these two lines for the majority of users makes it impossible for the other users to determine whether there was or was not a newline.

Imagine someone ran `cat` or `unrar` and the output bytes encoded for a file. If that file ended in a newline, you would ruin the file for them in a way that no one would ever be able to figure out. Text files always end with newlines in Linux. If a binary file's last byte were a newline, you could render the file unusable until someone discovered what had happened and why.